subsys/net/l2/ethernet/gptp/gptp_mi.c

```diff
@@ -772,47 +793,331 @@ static void gptp_mi_clk_slave_sync_state_machine(void)
 	}
 }
 
-static void gptp_mi_clk_master_sync_rcv_state_machine(void)
+static void gptp_mi_clk_master_sync_offset_state_machine(void)
+{
+	struct gptp_clk_master_sync_offset_state *state;
+	struct gptp_global_ds *global_ds;
+
+	state = &GPTP_STATE()->clk_master_sync_offset;
+	global_ds = GPTP_GLOBAL_DS();
+
+	switch (state->state) {
+	case GPTP_CMS_OFFSET_INITIALIZING:
+		state->rcvd_sync_receipt_time = false;
+		state->state = GPTP_CMS_OFFSET_INDICATION;
+		break;
+	case GPTP_CMS_OFFSET_INDICATION:
+		if (!state->rcvd_sync_receipt_time) {
+			break;
+		}
+
+		state->rcvd_sync_receipt_time = false;
+
+		if (global_ds->selected_role[0] == GPTP_PORT_PASSIVE) {
+			/* TODO Calculate real values for proper BC support */
+			memset(&global_ds->clk_src_phase_offset, 0x0,
+			       sizeof(struct gptp_scaled_ns));
+			global_ds->clk_src_freq_offset = 0;
+		} else if (global_ds->clk_src_time_base_indicator_prev
+			   != global_ds->clk_src_time_base_indicator) {
+			memcpy(&global_ds->clk_src_phase_offset,
+			       &global_ds->last_gm_phase_change,
+			       sizeof(struct gptp_scaled_ns));
+
+			global_ds->clk_src_freq_offset =
+				global_ds->last_gm_freq_change;
+		}
+
+		break;
+	default:
+		NET_ERR("Unrecognised state %d", state->state);
+		break;
+	}
+}
+
+#if defined(CONFIG_NET_GPTP_GM_CAPABLE)
+static inline void gptp_mi_setup_sync_send_time(void)
+{
+	struct gptp_clk_master_sync_snd_state *state;
+	struct gptp_global_ds *global_ds;
+	u64_t time_helper;
+
+	state = &GPTP_STATE()->clk_master_sync_send;
+	global_ds = GPTP_GLOBAL_DS();
+
+	time_helper = state->sync_send_time.low;
+
+	state->sync_send_time.low +=
+		global_ds->clk_master_sync_itv;
+
+	/* Check for overflow */
+	if (state->sync_send_time.low < time_helper) {
+		state->sync_send_time.high += 1;
+		state->sync_send_time.low =
+			UINT64_MAX - state->sync_send_time.low;
+	}
+}
+
+static void gptp_mi_set_ps_sync_cmss()
 {
+	struct gptp_clk_master_sync_snd_state *state;
+	struct gptp_global_ds *global_ds;
+	struct gptp_md_sync_info *sync_info;
+	u64_t current_time;
+
+	global_ds = GPTP_GLOBAL_DS();
+	state = &GPTP_STATE()->clk_master_sync_send;
+
+	sync_info = &state->pss_snd.sync_info;
+
+	state->pss_snd.local_port_number = 0;
+
+	current_time = gptp_get_current_master_time_nanosecond();
+
+	sync_info->precise_orig_ts.second = current_time / NSEC_PER_SEC;
+	sync_info->precise_orig_ts.nanosecond = current_time % NSEC_PER_SEC;
+
+	/* TODO calculate correction field properly, rate_ratio is also set to
+	 * zero instead of being copied from global_ds as it affects the final
+	 * value of FUP correction field.
+	 */
+	sync_info->follow_up_correction_field = 0;
+	sync_info->rate_ratio = 0;
+
+	memcpy(&sync_info->src_port_id.clk_id,
+	       GPTP_DEFAULT_DS()->clk_id,
+	       GPTP_CLOCK_ID_LEN);
+
+	sync_info->src_port_id.port_number = 0;
+
```

Comment:
Probably the empty lines here and few lines below could be removed when we set the fields of the same variable. Now it looks excessively spacy.

Reply:
That's right, changed